src/test/java/net/minecraftforge/debug/gameplay/loot/GlobalLootModifiersTest.java

```diff
@@ -0,0 +1,193 @@
+     * @author Draco18s
+     *
+     */
+    private static class SmeltingEnchantmentModifier extends LootModifier {
+        public SmeltingEnchantmentModifier(ResourceLocation name, ILootCondition[] conditionsIn) {
+            super(conditionsIn);
+
+        }
+
+        @Override
+        public List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+            ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
+            generatedLoot.forEach((stack) -> ret.add(smelt(stack, context)));
+            return ret;
+        }
+
+        private static ItemStack smelt(ItemStack stack, LootContext context) {
+            return context.getWorld().getRecipeManager().getRecipe(IRecipeType.SMELTING, new Inventory(stack), context.getWorld())
+                    .map(FurnaceRecipe::getRecipeOutput)
+                    .filter(itemStack -> !itemStack.isEmpty())
+                    .map(itemStack -> ItemHandlerHelper.copyStackWithSize(itemStack, stack.getCount() * itemStack.getCount()))
+                    .orElse(stack);
+        }
+
+        private static class Serializer extends LootModifier.Serializer<SmeltingEnchantmentModifier> {
+            @Override
+            public SmeltingEnchantmentModifier read(ResourceLocation name, JsonObject json, ILootCondition[] conditionsIn) {
+                return new SmeltingEnchantmentModifier(name, conditionsIn);
+            }
+        }
+    }
+
+    /**
+     * When harvesting blocks with bamboo, this modifier is invoked, via the silk_touch_bamboo loot_modifier json
+     * @author Draco18s
+     *
+     */
+    private static class SilkTouchTestModifier extends LootModifier {
+        public SilkTouchTestModifier(ResourceLocation name, ILootCondition[] conditionsIn) {
+            super(conditionsIn);
+        }
+
+        @Override
+        public List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+            ItemStack ctxTool = context.get(LootParameters.TOOL);
+            ItemStack fakeTool = ctxTool.copy();
+            fakeTool.addEnchantment(Enchantments.SILK_TOUCH, 1);
+            if(ItemStack.areItemsEqual(ctxTool, fakeTool) && EnchantmentHelper.getEnchantments(ctxTool).containsKey(Enchantments.SILK_TOUCH)) return generatedLoot;
+            LootContext.Builder builder = new LootContext.Builder(context.getWorld())
+                    .withParameter(LootParameters.BLOCK_STATE, context.get(LootParameters.BLOCK_STATE))
+                    .withRandom(context.getWorld().rand)
+                    .withParameter(LootParameters.POSITION, context.get(LootParameters.POSITION))
+                    .withParameter(LootParameters.TOOL, fakeTool)
+                    .withNullableParameter(LootParameters.BLOCK_ENTITY, context.get(LootParameters.BLOCK_ENTITY));
+            if(context.has(LootParameters.THIS_ENTITY)) {
+                builder.withParameter(LootParameters.THIS_ENTITY, context.get(LootParameters.THIS_ENTITY));
+            }
```

Comment:
This seems super bad, why are you rebuilding the entire context? How can you possibly know the context and what parts need copying?

Reply:
Again, test mod code, not shippable.

Reply:
And if there's a better way, I'd be happy to do it. This was what came up in the prior conversation. LootContext objects are `private final` and the context *has* to be cloned in order for the functionality to work, as the original context should not be modified (as it will get passed to future LootModifier handlers). If that means we need to patch in a Builder.FromContext of some kind, then fine, I'll do that.